src/stories/components/Modal/Modal.js

```diff
@@ -0,0 +1,84 @@
+import './styles.scss';
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { Fragment } from 'react';
+import { Button } from '../Button/Button';
+
+/**
+ * Component for Modal element.
+ *
+ * @component
+ * @param {node} children items of the Modal.
+ * @param {string} theme of the Modal Button's parent.
+ * @return {object} (
+ *   <Modal>
+ *      {children}
+ *   </Modal>
+ * )
+ */
+export const Modal = ({ children, theme, buttonText }) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  // const [modalClasses, setModalClasses] = useState('modal--content');
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+  };
+
+  return (
+    <Fragment>
+      <Button
+        modifierClasses={[
+          'button--small',
+          `button--${
+            (theme === 'light')?
+              'secondary':
+              'quaternary'
+          }`].join(' ').trim()
+        }
+        onClick={toggleModal}
+        text={buttonText}
+        isButton={true}/>
+      <div
+        hidden={!modalOpen}
+        id='modalPop'
+        className={['modal', `modal--${theme}`].join(' ').trim()}
+      >
+        <div
+          className={
+            (modalOpen) ?
+              'modal--content modal--open' :
+              'modal--content'}
+        >
+          <span
+            onClick={toggleModal}
+            className='modal--content--close'
+          />
+          {children}
+        </div>
```

Comment:
Semantically speaking this should be a button. You can have text inside of it, visually hide it via CSS and use the pseudo-element to include the icon. This is the code snippet I use:
```
display: block;
color: transparent;
font: 0/0 a;
height: 24px;
text-shadow: none;
width: 24px;
position: relative;
```
It visually hides the text, but keeps it accessible to screen readers. It's pretty neat! Let me know if you have any questions!

Reply:
@alecherryy That's a great idea, I did not know that!
I tried it but I wasn't able to hide the button; an empty block of sorts still appears.
![Screen Shot 2021-03-03 at 2 23 22 PM](https://user-images.githubusercontent.com/23450260/109880677-17861300-7c2c-11eb-9d82-6955fdb44487.png)